### PR TITLE
Add input/output length options to MLPerf / MMLU datasets + add JAX server-based profiling

### PR DESCRIPTION
**Pull request by jrplatin**
# Description

In this PR, I do the following:
* Add `input` / `output` lengths arguments to the MLPerf/MMLU datasets
* Add ability to start a JAX profiling server for the JAX path.  To use it, simply set the following env var: `USE_JAX_PROFILER_SERVER=True` .  You can also optionally set the `JAX_PROFILER_SERVER_PORT`.
    * This is useful in the case where you'd like to profile a cross section of a full E2E benchmark run with many prompts (instead of using the old method of setting `VLLM_TORCH_PROFILER_DIR` and running benchmarking with a single prompt).

In order to capture (cross-sectional) profiles:

1. Run your typical `vllm serve` or `offline_inference` cmd (making sure to set `USE_JAX_PROFILER_SERVER=True)
2. Run your benchmarking command (`python benchmark_serving.py...`)
3. Once the warmup has completed and your benchmark is running, start a new tensorboard instance with your `logdir` set to the desired output location of your logs (e.g. `tensorboard --logdir=profiles/llama3-mmlu/`
4. Open the tensorboard instance and navigate to the `profile` page (e.g. `http://localhost:6006/#profile`)
5. Click `Capture Profile` and, in the `Profile Service URL(s) or TPU name` box, enter `localhost:XXXX` where `XXXX` is your `JAX_PROFILER_SERVER_PORT` (default is `9999`):
image

6. Enter the desired amount of time you'd like to capture the profile for and then click `Capture`.  If everything goes smoothly, you should see the following, with your `logdir` being populated:
image

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have made or will make corresponding changes to any relevant documentation.

### CODE REVIEW

File: scripts/vllm/benchmarking/benchmark_dataset.py [hunk 1]
```diff
@@ -287,11 +288,15 @@ def sample(
             #     skip_min_output_len_check=output_len is not None,
             # ):
             #     continue
+            if input_len <= prompt_len:
```
Comment (bzgoogle) [inline scripts/vllm/benchmarking/benchmark_dataset.py]: may i know what does input_len refer to?  prompt_len is the # tokens in a prompt, what about input_len?

File: tpu_commons/worker/tpu_worker_jax.py [hunk 2]
```diff
@@ -72,6 +72,15 @@ def __init__(self,
 
         logger.info(f"Using devices: {self.devices}")
 
+        use_jax_profiler_server = os.getenv("USE_JAX_PROFILER_SERVER", False)
```
Comment (bzgoogle) [inline tpu_commons/worker/tpu_worker_jax.py:75]: is the new change still compatible with the old way of profiling? i.e. add --profile and it will dump to the pointer given to vllm serve

File: scripts/vllm/benchmarking/benchmark_dataset.py (hunk 1 above)
Comment (jrplatin) [reply scripts/vllm/benchmarking/benchmark_dataset.py]: It's the `prompt_len` yep, I just kept it as a `input_len` since that's the vLLM convention

File: tpu_commons/worker/tpu_worker_jax.py (hunk 2 above)
Comment (jrplatin) [reply tpu_commons/worker/tpu_worker_jax.py:75]: Yep, the old way will still work!
